sphinx/environment/__init__.py

```diff
@@ -788,3 +789,22 @@ def _traverse_toctree(
             if sub_docname not in traversed:
                 yield sub_parent, sub_docname
                 traversed.add(sub_docname)
+
+
+def _check_toc_parents(toctree_includes:  dict[str, list[str]]):
+    toc_parents: dict[str, list[str]] = {}
+    for parent, children in toctree_includes.items():
+        for child in children:
+            toc_parents.setdefault(child, []).append(parent)
+
+    for doc, parents in sorted(toc_parents.items()):
+        if len(parents) > 1:
+            logger.warning(
+                __("document is referenced in multiple toctrees: %s, "
+                   "selecting: %s <- %s"),
```

Comment:
We probably have to put translations for this (I'm not sure how).

Reply:
Apologies for not responding to this previously: what you have here is correct, I believe.

There are automated processes in the Sphinx continuous integration that scan for translation strings (as indicated by the use of the `__(...)` call), and that request translation of them in Transifex (a translation platform) -- and also that pull in the latest translations from there once available.